lib/puppet/http/service/ca.rb

```diff
@@ -10,7 +10,7 @@ def initialize(client, server, port)
   def get_certificate(name, ssl_context: nil)
     response = @client.get(
       with_base_url("/certificate/#{name}"),
-      headers: HEADERS,
+      headers: add_puppet_headers(HEADERS),
```

Comment:
Sorry I missed this earlier. The other methods in this file need to call `add_puppet_headers` like when downloading the CRL.